addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -2203,7 +2203,8 @@ export class Wysiwyg extends Component {
                     </div>
                 `).childNodes[0];
                 this.odooEditor.execCommand('insert', bannerElement);
-                setSelection(bannerElement.querySelector('.o_editor_banner > div'), 0);
+                this.odooEditor._activateContenteditable();
+                setSelection(bannerElement.querySelector('.o_editor_banner > div > p'), 0);
```

Comment:
sorry for the last minute input, i observed something which should be addressed as well. @dhba-odoo accessing the private method is not recommended, i have a suggestion which can be a workaround 
 1) by using 't-ref' to banner element and setting focus to it's <p>. 
2) or we can make '_activateContenteditable' a public method keeping it in utils. such that `wysiwyg` and `odooeditor` both can use it according to its convenience.
cc @Zynton what is your input on this ?

Reply:
It's indeed not ideal and I'm a little confused as to why it's being accessed anyway if the goal is just to set the focus. Wouldn't `bannerElement.focus()` suffice? If not, how come?

Reply:
Here, `bannerElement.focus()` will not work because the `contentEditable` attribute for the`bannerElement` is `false`.

Reply:
Ah I see that `contenteditable` is only set to true on the `div` that you want to edit (`.o_editor_banner > div`) because it's explicitely added as a content editable area:
```js
getContentEditableAreas: (...args) => {
    const areaEls = _getContentEditableAreas?.(...args) || [];
    const bannerEls = this.$editable[0].querySelectorAll('.o_editor_banner > div');
    return [...areaEls, ...bannerEls];
},
```
So indeed that only gets applied when `_activateContenteditable` is called, which means we need to make sure that gets called first... You could trigger a `mousedown` event for that but that's unclear and you'd have to deal with asynchronous stuff, so I'd argue in favor of making `_activateContenteditable` public. Although we can't change its name in stable so my suggestion is to add a method:
```js
activateContenteditable {
    this._activateContenteditable();
}
```
with a nice comment to justify it.

@sebgeelen is that ok for you?

Reply:
sure

Reply:
Done!